view/frontend/templates/js/replacejs.phtml

```diff
@@ -1470,6 +1499,19 @@ if (!$block->isOnPageFromWhiteList() && !$block->isMinicartEnabled()) { return;
         ////////////////////////////////////////////////////
         boltCheckoutSetup();
         ////////////////////////////////////////////////////
+        
+        <?php
+        if ($isLoadConnectJsDynamic) {
+        ?>
+        magentoCartDataListenerLoadConnectJs = function(data) {
+            if (data.items !== undefined && data.items.length > 0) {
+                insertConnectScript();
+            }            
+        }
+        customerData.get('cart').subscribe(magentoCartDataListenerLoadConnectJs);
```

Comment:
Why you prefer to use a new listener instead of using existing one?

Reply:
Cause here we only need to load connect js, and actually we expect it is executed only once, if the tag bolt-connect already exists, insertConnectScript won't load js repeatedly. Then the existing one does something else which is not needed.